app/src/main/java/bg/zahov/app/ui/settings/SettingsReusableFields.kt

```diff
@@ -0,0 +1,147 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import bg.zahov.fitness.app.R
+
+@Composable
+fun SettingsButton(text: String, onClick: () -> Unit) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .clickable { onClick() }
+            .padding(6.dp)
+    ) {
+        Text(
+            text = text,
+            modifier = Modifier
+                .padding(start = 12.dp),
+            fontSize = 16.sp,
+            color = colorResource(R.color.white)
+
+        )
+    }
+}
+
+@Composable
+fun SettingsRadioButton(
+    title: String,
+    text: String,
+    dialog: @Composable (onDismiss: () -> Unit) -> Unit,
+) {
+    var showDialog by remember { mutableStateOf(false) }
+
+    if (showDialog) {
+        dialog { showDialog = false }
+    }
+
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .clickable { showDialog = true }
+            .padding(vertical = 6.dp)
+    ) {
+        Text(
+            text = title,
+            modifier = Modifier
+                .padding(start = 12.dp),
+            fontSize = 16.sp,
+            color = colorResource(R.color.white)
+        )
+        Text(
+            text = text,
+            modifier = Modifier
+                .padding(top = 4.dp, start = 12.dp),
+            fontSize = 12.sp,
+            color = colorResource(R.color.text)
+        )
+    }
+}
+
+@Composable
+fun SettingsSwitchButton(
+    title: String,
+    text: String = "",
+    checked: Boolean,
+    onChecked: () -> Unit
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(vertical = 6.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Column {
+            Text(
+                text = title,
+                modifier = Modifier.padding(start = 12.dp),
+                fontSize = 16.sp,
+                color = colorResource(R.color.white)
+            )
+            Text(
+                text = text,
+                modifier = Modifier.padding(start = 16.dp, top = 4.dp),
+                fontSize = 12.sp,
+                color = colorResource(R.color.text)
+            )
+        }
+        Column(modifier = Modifier.fillMaxWidth()) {
+            Switch(
+                checked = checked,
+                onCheckedChange = { onChecked() },
+                modifier = Modifier
+                    .align(Alignment.End)
+                    .padding(end = 12.dp),
+
+                )
+        }
+    }
+}
+
+@Composable
+fun SettingsText(text: String) {
+    Text(
+        text = text,
+        fontSize = 20.sp,
+        color = Color.White,
+        modifier = Modifier.padding(top = 20.dp, start = 20.dp)
+    )
+}
+
+@Composable
+fun ColumnTemplate(items: @Composable (() -> Unit)) {
```

Comment:
Wouldn't it be simpler to just add dividers wherever they're needed in the original Column in SettingsScreen?

Reply:
that would happend in in each individual column so it less code for the SettingsContent

Reply:
My idea is to have 6 `Divider()` calls in `SettingsContent()` between the corresponding items, instead of using this function. That's because the code now is indented and we're calling additional functions which do the same thing of just adding a divider.

Reply:
I also use this so i don't have to copy - paste the same modifier

Reply:
That can be remedied by having a function for the divider itself.

Reply:
now is only a divider